rules/alerting.go

```diff
@@ -120,6 +120,8 @@ type AlertingRule struct {
 	annotations labels.Labels
 	// Time in seconds taken to evaluate rule.
 	evaluationDuration time.Duration
+	// Unix timestamp of last evaluation of rule
+	evaluationTimestamp time.Time
```

Comment:
The mutex has a comment `// Protects the below.`, but now there's already several fields above it in the struct that it also protects. Doesn't have to be done in this PR, but it would be good to reorganize this all to make it clear which fields the mutex protects.